### PR TITLE
Cleanup some more code

### PR DESCRIPTION
**Pull request by mvdan**

### CODE REVIEW

**Comment by buger** [comment]
👍

File: tyk_reverse_proxy_clone.go [hunk 1]
```diff
@@ -238,7 +236,7 @@ func TykNewSingleHostReverseProxy(target *url.URL, spec *APISpec) *ReverseProxy
 
 		// No override, and no load balancing? Use the existing target
 		targetToUse := target
-		if switchTargets {
```
Comment (lonelycode) [inline tyk_reverse_proxy_clone.go]: There isn't a test to cover this functionality, it looks ok, but we can't guarantee it works (Use case is: a path that is being rewritten has a hostname in it, and so needs re-parsing so the proxy target changes)

Comment (mvdan) [inline tyk_reverse_proxy_clone.go]: Agreed, will undo. This does change functionality somewhat, as empty strings will be treated differently.

Comment (mvdan) [inline tyk_reverse_proxy_clone.go]: @lonelycode done, PTAL

**Comment by mvdan** [comment]
It seems I can't mark the requested change as solved. I assume only you can do that.

**Pull request merged**